include/deal.II/distributed/cell_data_transfer.h

```diff
@@ -0,0 +1,346 @@
+        sum(const typename parallel::distributed::Triangulation<dim, spacedim>::
+              cell_iterator & parent,
+            const VectorType *input_vector)
+        {
+          typename VectorType::value_type sum = 0;
+
+          for (unsigned int child_index = 0; child_index < parent->n_children();
+               ++child_index)
+            sum +=
+              (*input_vector)[parent->child(child_index)->active_cell_index()];
+
+          return sum;
+        }
+
+        /**
+         * Evaluate data from @p input_vector on children of @p parent.
+         * Return mean value.
+         */
+        static typename VectorType::value_type
+        mean(const typename parallel::distributed::
+               Triangulation<dim, spacedim>::cell_iterator &parent,
+             const VectorType *                             input_vector)
+        {
+          return sum(parent, input_vector) / parent->n_children();
+        }
+      };
+
+      /**
+       * Constructor.
+       *
+       * @param[in] triangulation The triangulation on which all operations will
+       *   happen. At the time when this constructor is called, the refinement
+       *   in question has not happened yet.
+       * @param[in] transfer_variable_size_data Specify whether size of data to
+       *   transfer varies from cell to cell.
+       * @param[in] coarsening_strategy Function deciding which data to store on
+       *   a cell, whose children will get coarsened into.
+       */
+      CellDataTransfer(const parallel::distributed::Triangulation<dim, spacedim>
+                         &        triangulation,
+                       const bool transfer_variable_size_data = false,
+                       std::function<typename VectorType::value_type(
+                         const typename parallel::distributed::
+                           Triangulation<dim, spacedim>::cell_iterator &parent,
+                         const VectorType *input_vector)> coarsening_strategy =
+                         &CoarseningStrategies::check_equality);
+
+      /**
+       * Prepare the current object for coarsening and refinement. It registers
+       * the data transfer of @p in on the underlying triangulation. @p in includes
+       * data to be interpolated onto the new (refined and/or coarsened) grid.
+       */
+      void
+      prepare_for_coarsening_and_refinement(const VectorType &in);
+
+      /**
+       * Same as the function above, only for a list of vectors.
+       */
+      void
+      prepare_for_coarsening_and_refinement(
+        const std::vector<const VectorType *> &all_in);
+
+      /**
+       * Prepare the serialization of the given vector. The serialization is
+       * done by Triangulation::save(). The given vector needs all information
+       * on the locally active DoFs (it must be ghosted). See documentation of
+       * this class for more information.
+       */
+      void
+      prepare_serialization(const VectorType &in);
```

Comment:
this is the place where I would suggest to name the function `prepare_for_serialization()` (same below). Or have we made the same mistake in other places already?

Reply:
Different question: On the same object, can this function be called more than once with different vectors? I suspect the answer is "no", but it might be worth mentioning, with a hint towards the next function.

Reply:
We already named these function `prepare_serialization` in the `SolutionTransfer` class, so I adopted the interface.

You are right, the answer is 'no'. I updated the documentation: The function can be called once until transfer completed.